src/main.cpp

```diff
@@ -59,6 +60,11 @@ set<CWallet*> setpwalletRegistered;
 
 CCriticalSection cs_main;
 
+// This is for a non-Boost implementation of interruptible sleep. See MilliSleep in time.h/cpp. We initialize the globals
+// here.
+std::condition_variable sleep_interrupt;
```

Comment:
Should these maybe to into `time.cpp` instead, or is this an upstream thing?

Reply:
As long as we don't have a problem with global destruct order, I don't mind moving them.

Reply:
Upstream isn't doing a generalized interruptible sleep anymore...

Reply:
Done. Works fine.

Reply:
> Upstream isn't doing a generalized interruptible sleep anymore...

Does CThreadInterrupt not work?

Reply:
You are right. It is easier to just take that from upstream too. Done.